src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -527,6 +528,13 @@ private ReadQuery getSliceCommands(QueryOptions options, ColumnFilter columnFilt
         if (keys.isEmpty())
             return ReadQuery.empty(table);
 
+        if (restrictions.keyIsInRelation())
+        {
+            Guardrails.partitionKeysInSelect.guard(keys.size(),
+                                                   String.format("partition keys in IN clause on table %s", table.name),
```

Comment:
We can put the `partition keys in IN clause on table` message in the message provider at `Guardrails.partitionKeysInSelect`, so here we only need to pass the table name.

Reply:
👍